redash/handlers/metr_widgets.py

```diff
@@ -0,0 +1,34 @@
+
+from flask import request
+
+from redash import models
+from redash.handlers.base import BaseResource
+from redash.permissions import (
+    require_object_modify_permission,
+    require_permission
+)
+from redash.serializers import serialize_widget
+
+class metrWidgetTagsResource(BaseResource):
+    """
+    Resource for handling tags of a metrWidget.
+
+    This resource provides methods for retrieving and updating tags associated with a metrWidget.
+    """
+
+    @require_permission("edit_dashboard")
+    def post(self, widget_id):
+        widget = models.Widget.get_by_id_and_org(widget_id, self.current_org)
+        require_object_modify_permission(widget.dashboard, self.current_user)
+        widget_properties = request.get_json(force=True)
+        tags = widget_properties["tags"]
+
+        metrWidget=models.metrWidget.query.filter(models.metrWidget.widget==widget).first()
+        if metrWidget is None:
+            metrWidget=models.metrWidget(widget=widget, tags=tags)
+            models.db.session.add(metrWidget)
+        else:
+            metrWidget.tags = tags
+
```

Comment:
I'm not sure if this would work because I don't know SQLAlchemy a lot, but I think it would be easier to read without the else:

```suggestion
        if metrWidget is None:
            metrWidget=models.metrWidget(widget=widget)
            models.db.session.add(metrWidget)
        
        metrWidget.tags = tags
```

Reply:
Thanks, I believe that it is much more readable after i updated it[ in this commit](https://github.com/metr-systems/redash/pull/4/commits/fa3d1d79a3236ea521c555fd268cc62907dd697d), i don't really feel very comfortable removing the if-else because i don't want it to affect the tags twice to the instance but i honestly don't know if it is better than keeping the condition, what do you think about leaving it like that?

Reply:
thanks, this update is included[ in this commit ](https://github.com/metr-systems/redash/pull/4/commits/7f337577ff4cca5ee75b2645d8884daf85ab7abd)